src/renderer/components/screens/MainScreen.tsx

```diff
@@ -42,14 +44,46 @@ import SettingsStoreInstance, { useSettingsStore } from '../../stores/settings'
 import { Type } from '../../backend-com'
 import { InlineVerifiedIcon } from '../VerifiedIcon'
 import { SettingsProfileDialog } from '../dialogs/Settings-Profile'
+import UnreadBadge from '../UnreadBadge'
 
 const log = getLogger('renderer/main-screen')
 
 export default function MainScreen() {
   const [queryStr, setQueryStr] = useState('')
   const [queryChatId, setQueryChatId] = useState<null | number>(null)
   const [sidebarState, setSidebarState] = useState<SidebarState>('init')
-  const [showArchivedChats, setShowArchivedChats] = useState(false)
+  const [showArchivedChats, setShowArchivedChats] = useState<boolean>(false)
+  const [
+    haveOtherAccountsUnread,
+    setHaveOtherAccountsUnread,
+  ] = useState<boolean>(false)
+
+  useEffect(() => {
+    let updating = false
+    const update = () => {
+      if (updating) return
+      updating = true
+      BackendRemote.rpc.getAllAccountIds().then(accountIds => {
+        accountIds = accountIds.filter(id => id !== selectedAccountId())
+        for (const accountId of accountIds) {
+          BackendRemote.rpc.getFreshMsgs(accountId).then(ids => {
+            if (ids.length > 0) setHaveOtherAccountsUnread(true)
+          })
+        }
+      })
+      updating = false
+    }
+    update()
+
+    BackendRemote.rpc
+      .getAllAccountIds()
+      .then(accountIds =>
+        accountIds.map(id => onDCEvent(id, 'IncomingMsg', update))
+      )
+
+    return update
```

Comment:
there is already much happening in the MainScreen file, I propose you move the code to the unread badge component and rename that badge to sth like OtherAccountsUnreadMessagesBagde or sth that reflects that it shows that it's about wether the other accounts have unread messages.

I think the small runtime code duplication is not an issue, also we could add a timed debounce to limit how often it is updated. (instead of only making sure it is only run once at a time)